zellij-server/src/panes/terminal_character.rs

```diff
@@ -40,6 +42,13 @@ pub enum AnsiCode {
     NamedColor(NamedColor),
     RgbCode((u8, u8, u8)),
     ColorIndex(u8),
+
+    // underline styles
+    Underline,
+    Double,
+    Undercurl,
+    Underdotted,
+    Underdashed,
```

Comment:
Could we move these out to their own enum? I think it would be more relevant and less confusing.

Reply:
I added these to a new enum `AnsiStyledUnderline`. This required adding a few new fields and methods on `CharacterStyles`. Let me know if you were thinking of a different way to handle this.

Reply:
I think you meant `AnsiCode::Underline(Option<AnsiStyledUnderline>)`. If so, I did that and it's a much better solution.

Reply:
It also fixed the problem with single underlines not appearing.